core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/workflow/WorkflowCompiler.scala

```diff
@@ -100,6 +102,11 @@ class WorkflowCompiler(val workflowInfo: WorkflowInfo, val context: WorkflowCont
       }
       val amberOperator: OpExecConfig =
         o.operatorExecutor(OperatorSchemaInfo(inputSchemas, outputSchemas))
+      if (o.operatorInfo.operatorGroupName == OperatorGroupConstants.UDF_GROUP) {
+        // Currently, all UDFs are python operators.
+        // TODO: change this if we add UDFs for other languages.
+        pythonOperators.add(amberOperator)
+      }
```

Comment:
I don't feel using the group name is a good way to determine if it is a python operator. We might easily move python operators into different groups. For example, we will introduce "native Python Operator", such as the RedditSource operator, which is written in Python but belongs to the source operator group, user will drag the operator and use it as all other native python operators.

Reply:
So the reason for having this change is now we don't want to have any `IOperatorExecutor` on the controller side. Previously we check if `IOperatorExecutor` is a `PythonUDFOpExecV2`, but now we need another way of identifying python operators. I know this is not a good way to determine, so that's why I left a TODO there. We can refine it later.

Reply:
we are having some "native python" executors, for example reddit search operator and lambda expression operator, they will have a java OpDesc but executed with a PythonOpExec. they are likely not to be put under the `OperatorGroupConstants.UDF_GROUP` group. So I think we must find a better way to determine them now.